### DISCUSSION TITLE
I would really love the support for curly underline instead of a straight line

### DISCUSSION THREAD

**Discussion by duguyue100** [ghostty-org/ghostty#4132] [Feature Requests, Ideas] [answered]
Awesome defaults and documentation, it took me only 20 minutes or so to get what I wanted.

However, I would love to see the support for configuring underline style and color. If this is already part of the plan or you would also like this to happen, please support this ticket.

**Comment by mitchellh** [comment, marked as answer]
We already support curly underlines, they're a separate style has to be set by the running program. I don't think we should alter the default style. You can read more here (we follow Kitty's spec): https://sw.kovidgoyal.net/kitty/underlines/

**Reply by duguyue100** [reply]
This is a fair answer, thanks! I tried this in the terminal and got the expected results `echo -e '\e[4:3m\e[58:2::240:143:104mfoobarbaz\e[59m\e[4:0m\n'`, a curly underline with orange color.

~~However, I tried a few things in neovim and neovim+tmux. I still couldn't figure out how to make it work there. I found some discussion about kitty+neovim issues but they don't seem to work the same way as for kitty somehow.~~

~~Can you or someone give some pointers? How do I set the curly underline correctly in neovim+ghostty?~~

**Comment by duguyue100** [comment]
Just for future encounters' reference, I figured it out that the root cause is that `$TERM=xterm-256colors`, when I changed it to `xterm-ghostty`, it all works fine now.

**Reply by mitchellh** [reply]
For background on this: the reason is because xterm doesn’t support curly underlines. This is one of the downsides of using an incorrect TERM. :)

**Comment by benjypng** [comment]
For those using Ghostty with tmux, the following configuration works:

```
set -g default-terminal "xterm-256color"
set -as terminal-overrides ',*:Setulc=\E[58::2::%p1%{65536}%/%d::%p1%{256}%/%{255}%&%d::%p1%{255}%&%d%;m'
```

**Reply by luisdavim** [reply]
This used to work for me but now I get a dashed line instead of an undercul, I haven't changed my tmux config...

**Comment by luisdavim** [comment]
I have this in my `tmux` config:
```tmux
# Undercurl
if-shell '[[ "$(uname -o)" != "Android" ]]' {
  # set -g default-terminal tmux-256color
  set -g default-terminal "${TERM}"
  set -as terminal-overrides ',*256col*:RGB'
  set -as terminal-overrides ',*:sitm=\E[3m' # Italics support for older ncurses
  set -as terminal-overrides ',*:smxx=\E[9m' # Strikeout
  set -as terminal-overrides ',*:Smulx=\E[4::%p1%dm'  # undercurl support
  set -as terminal-overrides ',*:Setulc=\E[58::2::%p1%{65536}%/%d::%p1%{256}%/%{255}%&%d::%p1%{255}%&%d%;m'  # underscore colours - needs tmux-3.0
}
```
and it results in:

```sh
# tmux show -s
backspace C-?
buffer-limit 50
command-alias[0] split-pane=split-window
command-alias[1] splitp=split-window
command-alias[2] "server-info=show-messages -JT"
command-alias[3] "info=show-messages -JT"
command-alias[4] "choose-window=choose-tree -w"
command-alias[5] "choose-session=choose-tree -s"
copy-command ''
default-terminal xterm-ghostty
editor nvim
escape-time 10
exit-empty on
exit-unattached off
extended-keys off
extended-keys-format xterm
focus-events on
history-file ''
message-limit 1000
prefix-timeout 0
prompt-history-limit 100
set-clipboard on
terminal-overrides[0] linux*:AX@
terminal-overrides[1] *256col*:RGB
terminal-overrides[2] *:sitm=\\E[3m
terminal-overrides[3] *:smxx=\\E[9m
terminal-overrides[4] "*:Smulx=\\E[4::%p1%dm"
terminal-overrides[5] "*:Setulc=\\E[58::2::%p1%{65536}%/%d::%p1%{256}%/%{255}%&%d::%p1%{255}%&%d%;m"
terminal-overrides[6] *256col*:Tc
terminal-features[0] xterm*:clipboard:ccolour:cstyle:focus:title
terminal-features[1] screen*:title
terminal-features[2] rxvt*:ignorefkeys
user-keys
```

but instead of an undercul, when running:
```
echo -e '\e[4:3m\e[58:2::240:143:104mfoobarbaz\e[59m\e[4:0m\n'
```

I get the word underlined with a dashed line, same issue in `neovim`...
<img width="630" height="83" alt="image" src="https://github.com/user-attachments/assets/a2cf6113-1dfd-466c-ada6-f428e1da7997" />

The same config works in `kitty`:
<img width="628" height="79" alt="image" src="https://github.com/user-attachments/assets/31be39bc-b4d1-4392-aa9e-703a36494bd3" />

So I think this is a regression @mitchellh .

**Reply by luisdavim** [reply]
it seems like the issue is the font size, it seems like with relatively small font sizes, the undercurl gets trimmed and we only see the top of it, making it look like a dashed line.
But it only seems to display the undercurl correctly at ridiculously large font sizes, I'm using:
```
# Fonts
font-family = JetBrainsMono Nerd Font Mono
font-family-bold = JetBrainsMono NFM Bold
font-family-italic = JetBrainsMono NFM Italic
font-family-bold-italic = JetBrainsMono NFM Bold Italic

font-size = 14
font-thicken = true
```

**Reply by gcgbarbosa** [reply]
> ```shell
> set -g default-terminal "${TERM}"
>   set -as terminal-overrides ',*256col*:RGB'
>   set -as terminal-overrides ',*:sitm=\E[3m' # Italics support for older ncurses
>   set -as terminal-overrides ',*:smxx=\E[9m' # Strikeout
>   set -as terminal-overrides ',*:Smulx=\E[4::%p1%dm'  # undercurl support
>   set -as terminal-overrides ',*:Setulc=\E[58::2::%p1%{65536}%/%d::%p1%{256}%/%{255}%&%d::%p1%{255}%&%d%;m'  # underscore colours - needs tmux-3.0
> ```

This helped me immensely. Thank you!

**Comment by luisdavim** [comment]
I've started a new discussion for this: https://github.com/ghostty-org/ghostty/discussions/9384
